src/Message/PurchaseRequest.php

```diff
@@ -70,11 +70,14 @@ private function sign($params, $api_key) {
 
 	public function sendData($data)
     {
+		// FIXME: You're not actually sending any data here.
```

Comment:
Case in point is Helcim, where the user really can fiddle with the amount, as it is POSTed direct from a form on the application to the remote gateway, with no signing of the amount, which is kind of crazy. The application needs to fetch the transaction from the gateway at the end and check all the details before it can be considered accepted and valid.

That one is an example of yet another redirect method - a DIRECT POST, with no advance notice to the gateway that the user is coming over with a transaction.